### PR TITLE
Remove +kubebuilder:default:=true annotation from MonitoringComponent in vz CRD

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

This pull request removes the **+kubebuilder:default:=true** annotation from the MonitoringComponent type in the Verrazzano CRD.  With the default value annotation, the enabled value for the VMI components in a Verrazzano custom resource would always show a value of true once applied.

For example, the following Verrazzano custom resource spec:
```
spec:
  components:
    elasticsearch:
      enabled: false
```
when applied would end up as follows with enabled having a value of true
```
spec:
  components:
    elasticsearch:
      enabled: true
```

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/installjob/install_config.go [hunk 1]
```diff
@@ -673,10 +674,12 @@ func getVMIInstallArgs(vzSpec *installv1alpha1.VerrazzanoSpec) []InstallArg {
...
-		vmiArgs = append(vmiArgs, InstallArg{
-			Name:  esEnabledValueName,
-			Value: strconv.FormatBool(vzSpec.Components.Elasticsearch.Enabled),
-		})
+		if vzSpec.Components.Elasticsearch.Enabled != nil {
+			vmiArgs = append(vmiArgs, InstallArg{
+				Name:  esEnabledValueName,
+				Value: strconv.FormatBool(*vzSpec.Components.Elasticsearch.Enabled),
+			})
+		}
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/installjob/install_config.go:682]: We don't want to append the install args if ES is not enabled, do we?

Comment (pfmackin) [reply platform-operator/controllers/verrazzano/installjob/install_config.go:682]: The enabled for ES should default to true.  Is that still the case?  We don't wan the user to have to explicitly specify true if they added install args.

Comment (bobdonat) [reply platform-operator/controllers/verrazzano/installjob/install_config.go:682]: The install arg will be appended if the enabled flag is specified in the vz CR. 
The default is still true for ES unless overridden by profile settings.

Comment (bobdonat) [reply platform-operator/controllers/verrazzano/installjob/install_config.go:682]: The install behavior is unchanged with these changes.

**Review by pfmackin — APPROVED** [review]

**Pull request merged**